sogs/model.py

```diff
@@ -288,6 +314,16 @@ def __init__(self, row=None, *, id=None, session_id=None, autovivify=True, touch
             if touch:
                 self._touch()
 
+    @property
+    def rooms(self):
+        result = list()
+        """ return the rooms this user is in by their id """
+        with db.conn as conn:
+            rows = conn.execute("SELECT room FROM room_users WHERE user = ?", (self.id,))
+            for row in rows:
+                result.append(row[0])
```

Comment:
No need to wrap a single SELECT in a transaction (aside from transaction overhead it also makes it difficult to call this if you are already in a transaction).

```suggestion
        rows = db.conn.execute("SELECT room FROM room_users WHERE user = ?", (self.id,))
        result = [row[0] for row in rows]
```

But I wonder if this wouldn't be more versatile if it returned Rooms objects instead of ids:
```suggestion
        rows = db.conn.execute("""
            SELECT room.* FROM room_users JOIN rooms ON room_users.room = room.id
            WHERE user = ?
            """,
            (self.id,)
        )
        result = [Room(row=row) for row in rows]
```